lm_eval/models/translators.py

```diff
@@ -44,18 +44,30 @@ def generate_until(self, reqs: list[Instance]) -> list[str]:
                 "target_lang": self.target_lang,
                 "model": self.model
             }
-
-            response = requests.post(self.endpoint,
-                                     headers=self.headers,
-                                     json=data)
-
-            if response.status_code != 200:
-                print(f"An error occurred while translating: {response.text}")
-                raise Exception(response.text)
-
-            result = response.json()
-            results.append(result['translations'][0]['text'])
-
+            max_attempts = 10
+            attempts = 0
+            while attempts < max_attempts:
+                try:
+                    response = requests.post(self.endpoint,
+                                             headers=self.headers,
+                                             json=data)
+
+                    if response.status_code != 200:
```

Comment:
I think `raise_for_status()` is good if you expect it to only have 4XX and 5XX errors.
But if it's possible to have 2XX other than 200, you can ignore it.
https://requests.readthedocs.io/en/latest/user/quickstart/#response-status-codes